### ISSUE TITLE
Stats: Missing empty states

### ISSUE THREAD

**Issue by a8ck3n** [Automattic/wp-calypso#71746]
The logic for the empty states doesn't take into account the visible data range.

**Missing empty state:**

https://user-images.githubusercontent.com/40267301/210933601-bef55eea-1a5e-43f5-871d-2ea82588c0c7.mp4

If we zoom out (ie: larger viewport) we see there is one day with data which is probably the cause of this.

**Larger chart view:**
![Image](https://user-images.githubusercontent.com/40267301/210933314-4ec25f34-620f-4e16-8586-22a9dfb76a2e.png)

As a user, I'd expect to see an empty state message if the chart was empty (as seen in the video) so the state should probably be fixed to match the currently visible date range.

Discovered when testing #71745.

**Comment by a8ck3n** [comment]
Did some digging today and found the cause of this bug. In the following code snippet, `nextVals` needs to be calculated based on `nextData` to properly reflect the visible data. If we base it on the full data set, we don't properly detect the empty state and we set the `yMax` incorrectly.

```
// Memoize data calculations to avoid performing them too often.
	const { chartData, isEmptyChart, yMax } = useMemo( () => {
		const nextData = data.length <= maxBars ? data : data.slice( 0 - maxBars );
		const nextVals = data.map( ( { value } ) => value );

		return {
			chartData: nextData,
			isEmptyChart: Boolean( ! nextVals.some( ( a ) => a > 0 ) ),
			yMax: getYAxisMax( nextVals ),
		};
	}, [ data, maxBars, hasResized ] );
```

https://github.com/Automattic/wp-calypso/blob/137daeb7b3d1e654d42dc7651c7497a4a69e8626/client/components/chart/index.jsx#L113

**Comment by a8ck3n** [comment]
Addressed in #71790.
